addons/misra.py

```diff
@@ -2740,36 +2740,70 @@ def misra_12_3(self, data):
                     prev = prev.previous
 
     def misra_12_4(self, cfg):
-        for expr in cfg.tokenlist:
+        if not cfg.tokenlist:
+            return
+        expr = cfg.tokenlist[0]
+        while expr:
             if not expr.astOperand2 or not expr.astOperand1:
+                expr = expr.next
```

Comment:
Refactoring:

Adding `expr = expr.next` before each continue is unfortunate as it's much copy paste and very easy to forget to add it somewhere. so an alternative could be:

    expr_next = cfg.tokenlist[0]
    while expr_next:
        expr = expr_next
        expr_next = expr.next